proxysql_tools/galera.py

```diff
@@ -1,44 +1,49 @@
+from proxysql_tools import log
 from proxysql_tools.entities.galera import (
-    GaleraNode, LOCAL_STATE_SYNCED, LOCAL_STATE_DONOR_DESYNCED,
-    CLUSTER_STATUS_PRIMARY
+    LOCAL_STATE_SYNCED,
+    LOCAL_STATE_DONOR_DESYNCED
 )
 from proxysql_tools.entities.proxysql import BACKEND_STATUS_ONLINE
 from proxysql_tools.managers.galera_manager import (
-    GaleraManager, GaleraNodeNonPrimary, GaleraNodeUnknownState
+    GaleraManager,
+    GaleraNodeNonPrimary,
+    GaleraNodeUnknownState
+)
+from proxysql_tools.managers.proxysql_manager import (
+    ProxySQLManager,
+    ProxySQLAdminConnectionError
 )
-from proxysql_tools.managers.proxysql_manager import ProxySQLManager
 
 
-def register_cluster_with_proxysql(proxy_host, proxy_admin_port,
-                                   proxy_admin_user, proxy_admin_pass,
-                                   hostgroup_writer, hostgroup_reader,
-                                   cluster_host, cluster_port, cluster_user,
-                                   cluster_pass):
+def register_cluster_with_proxysql(config):
     """Register a Galera cluster within ProxySQL. The nodes in the cluster
     will be distributed between writer hostgroup and reader hostgroup.
 
-    :param str proxy_host: The ProxySQL host.
-    :param int proxy_admin_port: The ProxySQL admin port.
-    :param str proxy_admin_user: The ProxySQL admin user.
-    :param str proxy_admin_pass: The ProxySQL admin password.
-    :param int hostgroup_writer: The ID of writer ProxySQL hostgroup.
-    :param int hostgroup_reader: The ID of reader ProxySQL hostgroup.
-    :param str cluster_host: Hostname of a node in Galera cluster.
-    :param int cluster_port: Port of a node in Galera cluster.
-    :param str cluster_user: MySQL username of a user in Galera cluster.
-    :param str cluster_pass: MySQL password of a user in Galera cluster.
+    :param ConfigParser.ConfigParser config: The config object.
     :return bool: Returns True on success, False otherwise.
```

Comment:
```
    :return: Returns True on success, False otherwise.
    :rtype: bool
```
Looks better.
new https://www.dropbox.com/s/8967gfqvcxlsf19/Screenshot%202017-03-06%2021.15.41.png?dl=0
vs
old https://www.dropbox.com/s/5ep24h4pttfpxx1/Screenshot%202017-03-06%2021.16.01.png?dl=0